### DISCUSSION TITLE
Does it work with DSM6 too?

### DISCUSSION THREAD

**Discussion by ruberwm** [weltenwort/frigate-synology-dsm7#94] [General]
Hi,
I have a ds1515+ still on DSM6 at the moment.
I want to move to DSM7 eventually but right now I need to stick to 6 and I'd like to install frigate.
I am looking for a stable version of frigate.

Does this version work with DSM6 too?
If yes, generally speaking (dsm7 and 6) what happens in case I restart the nas? Does frigate restart automatically or do I need to manually (or by a task schedule) do something?
I know that docker automatically restarts and I have other apps on docker that restart flawlessly, I am not sure about frigate. :)

Thank you. Have a nice day.

**Comment by weltenwort** [comment]
Hi @ruberwm :wave:

> Does this version work with DSM6 too?

I don't have any way to test with DSM6 myself, but the comment in https://github.com/blakeblackshear/frigate/discussions/2426#discussioncomment-1901394 suggests that it works even without the fix that this image adds. Maybe the DSM6 kernel has been compiled with `udev` support. :shrug:

> [...] generally speaking (dsm7 and 6) what happens in case I restart the nas?

For me (on DSM7) the docker engine automatically starts the container again after boot, so no manual intervention is required.

**Reply by ruberwm** [reply]
Thank you @weltenwort :)

> I don't have any way to test with DSM6 myself, but the comment in https://github.com/blakeblackshear/frigate/discussions/2426#discussioncomment-1901394 suggests that it works even without the fix that this image adds. Maybe the DSM6 kernel has been compiled with udev support. 🤷

Ok, since I'd like to move to DSM7 and I'd like to keep consinstency, I'll give a try with your image :)

> For me (on DSM7) the docker engine automatically starts the container again after boot, so no manual intervention is required.

Perfect!

**Reply by weltenwort** [reply]
let me know how it works out on DSM6

**Comment by Floriszz** [comment]
I run Frigate latest stable(0.11.1-2EADA21) on my Synology NAS DS918+ with DSM6x(latest) and Google Coral USB stick flawless for couple of months now. (waiting on Frigate 0.12.x stable before moving to DSM 7.x)
I don't use cli/ssh and/or Docker compose. Just the Docker run commands through DSM Task Scheduler to install/run/update.
After reboot or cold start(we had power outage last week) everything comes up automatically.

**Reply by mx-vatte** [reply]
Could you please share the commands you used with the DSM task scheduler?

Thx in advance,
Vatte

**Reply by Floriszz** [reply]
sure!
I started with the example on: https://docs.frigate.video/frigate/installation 
(under; "If you can't use docker compose, you can run the container with something similar to this:")
In this sample I had to change a couple of lines(suggested to change this in the above source url as well through github) **Bold** marked.

docker run -d \
  --name frigate \
  --restart=unless-stopped \
  --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
  --device /dev/dri/renderD128 \
  --shm-size=64m \
  --**privileged** \
  -v /path/to/your/storage:/media/frigate \
  -v /path/to/your/config.yml:/config/config.yml:ro \
  -v /etc/localtime:/etc/localtime:ro \
  -**v /dev/bus/usb:/dev/bus/usb** \
  -**v /path/to/your/db:/db** \
  -e FRIGATE_RTSP_PASSWORD='password' \
  -p **5055**:5000 \
  -p 1935:1935 \
  blakeblackshear/frigate:stable

**Reply by mx-vatte** [reply]
The installation by DSM task scheduler seems to work  but the container kills with the following protocol.

docker run -d \
--name frigate \
--restart=unless-stopped \
--mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
--device /dev/bus/usb:/dev/bus/usb \
--shm-size=128m \
--privileged \
-v /volume1/docker/frigate/storage:/media/frigate \
-v /volume1/docker/frigate/config.yml:/config/config.yml:ro \
-v /etc/localtime:/etc/localtime:ro \
-v /dev/bus/usb:/dev/bus/usb \
-v /volume1/docker/frigate/db:/db \
-e FRIGATE_RTSP_PASSWORD='1234567890' \
-p 5055:5055 \
-p 1935:1935 \
blakeblackshear/frigate:stable

`<html>
<body>


2023-03-22T16:29:56.174429598Z | stdout | [cont-finish.d] done.
-- | -- | --
2023-03-22T16:29:56.174001546Z | stdout | [cont-finish.d] executing container finish scripts...
2023-03-22T16:29:56.140310191Z | stdout | [cmd] \ exited 127
2023-03-22T16:29:56.138940334Z | stderr | ifelse: fatal: unable to exec \: No such file or directory
2023-03-22T16:29:56.110303756Z | stdout | [services.d] done.
2023-03-22T16:29:56.094226713Z | stdout | [services.d] starting services
2023-03-22T16:29:56.093529609Z | stdout | [cont-init.d] done.
2023-03-22T16:29:56.092960856Z | stdout | [cont-init.d] executing container initialization scripts...
2023-03-22T16:29:56.092354763Z | stdout | [fix-attrs.d] done.
2023-03-22T16:29:56.091746270Z | stdout | [fix-attrs.d] applying ownership & permissions fixes...
2023-03-22T16:29:56.090860035Z | stdout | [s6-init] ensuring user provided files have correct perms...exited 0.
2023-03-22T16:29:56.061358953Z | stdout | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.


</body>
</html>`

Any idea where I've made a mistake?

**Reply by Floriszz** [reply]
Did you create 3 folders for the Docker image on you nas under the ' frigate'  folder?
![image](https://user-images.githubusercontent.com/19168937/226986346-3138f68f-da34-4e73-aa20-053269c164c8.png)

**Reply by mx-vatte** [reply]
I tried different structures already. Everytime the same error.
<img width="100" alt="image" src="https://user-images.githubusercontent.com/123895372/226991210-5752a2ee-473f-4264-9ea5-1af69a1e2a79.png">

**Reply by Floriszz** [reply]
@mx-vatte , seems you run in the same issue I broke my head over for weeks, why I could not get the container to stay running.
Check the ports you configured! The second port is the internal port for the container. You are not supposed to change this port.
It should be 5000! See my example

**Reply by Floriszz** [reply]
Also check where you put the config.yml. 
In your example with; -v /volume1/docker/frigate/config.yml:/config/config.yml:ro
the config.yml should be in the root of the 'frigate' folder. If you want the file in a folder 'config' it should be;
-v /volume1/docker/frigate/config/config.yml:/config/config.yml:ro or
-v /volume1/docker/frigate/storage/config/config.yml:/config/config.yml:ro

**Comment by mx-vatte** [comment]
Thx for your reply. In the meantime I got frigate running by the standard way with SSH onto the Diskstation. Just waiting for the coral stick to arrive to implement more cameras. The absolute basics are explained in this [discussion](https://github.com/weltenwort/frigate-synology-dsm7/discussions/91).

I think with my knowledge now and though the right config etc. It would also run by DSM task scheduler method.

**Reply by Floriszz** [reply]
thanks for the link. I think we can help others while using the task scheduler is super easy and saver compared to enable SSH and using command line commands.
So I posted my suggestion in the link you shared above. Hopefully it will be added to the general documentation of Frigate and can help many others.
So happy with the Object recognition instead of the motion sensing only in/with Surveillance station.
